src/test/groovy/hudson/plugins/gradle/injection/InjectionConfigTest.groovy

```diff
@@ -14,6 +15,40 @@ class InjectionConfigTest extends BaseGradleInjectionIntegrationTest {
     @Shared
     FilenameFilter injectionConfigXmlFilter = { _, name -> name == "hudson.plugins.gradle.injection.InjectionConfig.xml" }
 
+    @Unroll
+    def "sets showLegacyConfigurationWarning to true if any of legacy env variables is set"() {
+        given:
+        def env = new EnvironmentVariablesNodeProperty()
+        env.getEnvVars().put("TEST", "true")
```

Comment:
do you need this env var?

Reply:
I think yes, just to highlight that the test and the code are checking for specific vars but users can still have whatever other vars to be set.